pkgs/checks/lib/src/extensions/iterable.dart

```diff
@@ -13,6 +13,74 @@ extension IterableChecks<T> on Subject<Iterable<T>> {
   Subject<T> get last => has((l) => l.last, 'last element');
   Subject<T> get single => has((l) => l.single, 'single element');
 
+  /// Checks that the number of values in [Iterable] is exactly [length]
+  void hasLength(int length) {
+    if (length < 0) {
+      throw ArgumentError('length must be greater than 0');
+    }
+    context.expect(() => ['has length <$length>'], (actual) {
+      if (actual.length == length) return null;
+      return Rejection(
+          which: ['does not have length $length but <${actual.length}>']);
+    });
+  }
+
+  /// Checks that the number of values in [Iterable] is greater than [length]
+  void hasLengthGreaterThan(int length) {
+    if (length < 0) {
+      throw ArgumentError('length must be greater than 0');
+    }
+    context.expect(() => ['has length greater than <$length>'], (actual) {
+      if (actual.length > length) return null;
+      return Rejection(which: [
+        'does not have length greater than $length but <${actual.length}>'
+      ]);
+    });
+  }
+
+  /// Checks that the number of values in [Iterable] is greater than or equal to
+  /// [length]
+  void hasLengthGreaterThanOrEqualTo(int length) {
+    if (length < 0) {
+      throw ArgumentError('length must be greater than 0');
+    }
+    context.expect(() => ['has length greater than or equal to <$length>'],
+        (actual) {
+      if (actual.length >= length) return null;
+      return Rejection(which: [
+        'does not have length greater than or equal to $length but <${actual.length}>'
+      ]);
+    });
+  }
+
+  /// Checks that the number of values in [Iterable] is less than [length]
+  void hasLengthLessThan(int length) {
+    if (length < 0) {
+      throw ArgumentError('length must be greater than 0');
+    }
+    context.expect(() => ['has length less than <$length>'], (actual) {
```

Comment:
I was following the current style leading to this error:

```
Expected: a List<dynamic> that:
  has length less than <0>
Actual: []
Which: does not have length less than 0 but <0>
```

But, wouldn't it be easier to understand if the error message would use math symbols?

```
Expected: a List<dynamic> that:
  has length < 0
Actual: []
Which: does not have length < 0 but length == 0
```

Reply:
> But, wouldn't it be easier to understand if the error message would use comparison symbols?

I'll take a look at the existing extensions, this does look like an easier to read formatting.